### ISSUE TITLE
Jackal does not respond to registration request

### ISSUE THREAD

**Issue by ThomasLeister** [ortuman/jackal#75]
Hello,

I've set up Jackal on an entirely new machine (Debian Stretch) with PostgreSQL. Unfortunately I'm not able to create a new account on my server. Both Pidgin and Conversations for Android get stuck and neither the server nor the client is responding after a certain point. 

```
Feb 07 17:23:52 jackal jackal[1619]: 2019-02-07 17:23:52 ℹ️ [INF] c2s/server:147 - registered c2s stream... (id: c2s:default:3)
Feb 07 17:23:52 jackal jackal[1619]: 2019-02-07 17:23:52 🔍 [DBG] session/session:219 - RECV(c2s:default:3): <stream:stream version="1.0" xml:lang="en" xmlns:stream="http://etherx.jabber.org/streams" xmlns="jabber:client" to="jackal.650thz.de"/>
Feb 07 17:23:52 jackal jackal[1619]: 2019-02-07 17:23:52 🔍 [DBG] session/session:182 - SEND(c2s:default:3): <?xml version="1.0"?><stream:stream xmlns="jabber:client" xmlns:stream="http://etherx.jabber.org/streams" id="46407619-0aed-471d-ba71-67ba9f7aed9d" from="jackal.650thz.de" version="1.0"><stream:features xmlns:stream="http://etherx.jabber.org/streams" version="1.0"><starttls xmlns="urn:ietf:params:xml:ns:xmpp-tls"><required/></starttls></stream:features>
Feb 07 17:23:52 jackal jackal[1619]: 2019-02-07 17:23:52 🔍 [DBG] session/session:219 - RECV(c2s:default:3): <starttls xmlns="urn:ietf:params:xml:ns:xmpp-tls"/>
Feb 07 17:23:52 jackal jackal[1619]: 2019-02-07 17:23:52 🔍 [DBG] session/session:209 - SEND(c2s:default:3): <proceed xmlns="urn:ietf:params:xml:ns:xmpp-tls"/>
Feb 07 17:23:52 jackal jackal[1619]: 2019-02-07 17:23:52 ℹ️ [INF] c2s/in:487 - secured stream... id: c2s:default:3
Feb 07 17:23:52 jackal jackal[1619]: 2019-02-07 17:23:52 🔍 [DBG] session/session:219 - RECV(c2s:default:3): <stream:stream version="1.0" xml:lang="en" xmlns:stream="http://etherx.jabber.org/streams" xmlns="jabber:client" to="jackal.650thz.de"/>
Feb 07 17:23:52 jackal jackal[1619]: 2019-02-07 17:23:52 🔍 [DBG] session/session:182 - SEND(c2s:default:3): <?xml version="1.0"?><stream:stream xmlns="jabber:client" xmlns:stream="http://etherx.jabber.org/streams" id="50fd5c57-1857-47af-b253-3dc222122141" from="jackal.650thz.de" version="1.0"><stream:features xmlns:stream="http://etherx.jabber.org/streams" version="1.0"><mechanisms xmlns="urn:ietf:params:xml:ns:xmpp-sasl"><mechanism>PLAIN</mechanism><mechanism>DIGEST-MD5</mechanism><mechanism>SCRAM-SHA-1</mechanism><mechanism>SCRAM-SHA-1-PLUS</mechanism><mechanism>SCRAM-SHA-256</mechanism><mechanism>SCRAM-SHA-256-PLUS</mechanism><mechanism>SCRAM-SHA-512</mechanism><mechanism>SCRAM-SHA-512-PLUS</mechanism></mechanisms><register xmlns="http://jabber.org/features/iq-register"/></stream:features>
Feb 07 17:23:52 jackal jackal[1619]: 2019-02-07 17:23:52 🔍 [DBG] session/session:219 - RECV(c2s:default:3): <iq type="get" id="fTpaDp5chmgJ" to="jackal.650thz.de"><query xmlns="jabber:iq:register"/></iq>
```

**Jackal Version**: 0.4.8
**Database Backend**: PostgreSQL 9.6+181+deb9u2
TLS CA: Let's Encrypt

Configuration:

```
pid_path: /var/run/jackal/jackal.pid

debug:
  port: 6060

logger:
  level: debug
  log_path: /var/log/jackal/jackal.log

#storage:
#  type: mysql
#  mysql:
#    host: 127.0.0.1:3306
#    user: jackal
#    password: password
#    database: jackal
#    pool_size: 16

storage:
  type: pgsql
  pgsql:
    host: 127.0.0.1:5432
    user: jackal
    password: [password]
    database: jackal


#  cluster:
#  name: node1
#  port: 5010
#  hosts: [127.0.0.1:5009, 127.0.0.1:5011]

router:
  hosts:
    - name: jackal.650thz.de
      tls:
        privkey_path: "/etc/jackal/privkey.pem"
        cert_path: "/etc/jackal/fullchain.pem"

modules:
  enabled:
    - roster           # Roster
    - last_activity    # XEP-0012: Last Activity
    - private          # XEP-0049: Private XML Storage
    - vcard            # XEP-0054: vcard-temp
    - registration     # XEP-0077: In-Band Registration
    - version          # XEP-0092: Software Version
    - blocking_command # XEP-0191: Blocking Command
    - ping             # XEP-0199: XMPP Ping
    - offline          # Offline storage

  mod_roster:
    versioning: true

  mod_offline:
    queue_size: 2500

  mod_registration:
    allow_registration: yes
    allow_change: yes
    allow_cancel: yes

  mod_version:
    show_os: true

  mod_ping:
    send: no
    send_interval: 60

c2s:
  - id: default

    connect_timeout: 5
    max_stanza_size: 65536
    resource_conflict: replace  # [override, replace, reject]

    transport:
      type: socket # websocket
      bind_addr: 0.0.0.0
      port: 5222
      keep_alive: 120
      # url_path: /xmpp/ws

    compression:
      level: default

    sasl:
      - plain
      - digest_md5
      - scram_sha_1
      - scram_sha_256
      - scram_sha_512

#s2s:
#    dial_timeout: 15
#    dialback_secret: s3cr3tf0rd14lb4ck
#    max_stanza_size: 131072
#
#    transport:
#      bind_addr: 0.0.0.0
#      port: 5269
#      keep_alive: 600
```

Please let me know if you need any more information :-)

**Comment by ortuman** [comment]
I'll check it ASAP. Thank you once more for reporting. ;)

**Comment by ortuman** [comment]
Shoud be fixed in [0.4.9](https://github.com/ortuman/jackal/releases/tag/v0.4.9) release. 

Please, confirm if it works as it should 😄

**Comment by ThomasLeister** [comment]
Yes, it does! :-) Thanks for the quick fix!
